src/compiler/checker.ts

```diff
@@ -9583,17 +9903,23 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             function serializeAsFunctionNamespaceMerge(type: Type, symbol: Symbol, localName: string, modifierFlags: ModifierFlags) {
                 const signatures = getSignaturesOfType(type, SignatureKind.Call);
                 for (const sig of signatures) {
+                    context.approximateLength += 1; // ;
                     // Each overload becomes a separate function declaration, in order
                     const decl = signatureToSignatureDeclarationHelper(sig, SyntaxKind.FunctionDeclaration, context, { name: factory.createIdentifier(localName) }) as FunctionDeclaration;
                     addResult(setTextRange(context, decl, getSignatureTextRangeLocation(sig)), modifierFlags);
                 }
                 // Module symbol emit will take care of module-y members, provided it has exports
                 if (!(symbol.flags & (SymbolFlags.ValueModule | SymbolFlags.NamespaceModule) && !!symbol.exports && !!symbol.exports.size)) {
                     const props = filter(getPropertiesOfType(type), isNamespaceMember);
-                    serializeAsNamespaceDeclaration(props, localName, modifierFlags, /*suppressNewPrivateContext*/ true);
+                    context.approximateLength += localName.length;
+                    serializeAsNamespaceDeclaration(props, factory.createIdentifier(localName), modifierFlags, /*suppressNewPrivateContext*/ true);
                 }
             }
 
+            function createTruncationStatement(dotDotDotText: string): Statement {
+                return factory.createExpressionStatement(factory.createIdentifier(dotDotDotText));

```

Comment:
You should probably produce a `/* ... */;` (empty statement with a synthetic leading comment) result here when `IgnoreErrors` isn't passed, similar to other truncation sites. I understand it's not used internally today, but API consumers (and quickfixes that use the formatter) get weird behaviors when they try to roundtrip these non-identifier identifiers, so it's best to limit them to cases where we're planning on throwing an (ignored) error anyway.

Also, shouldn't this set the `truncated` out param inside this function? Or is that being done at all use-sites right now, instead? I think it'd be a bit cleaner to try and keep it inside all the factories making the different truncation statements, if possible.

Reply:
1. I'm not sure what you meant by `ignoreErrors`, I couldn't find it. But IIUC the places that were already doing truncation checked if the truncation flag was disabled (which it will be for scenarios like quickfixes) as a signal to use a `/* ... */` instead of a bad identifier name, so I added that.

2. The places that were already doing truncation don't all use separate functions to construct truncation nodes, so that wouldn't work unless I refactored all of those places.
I also considered setting this inside `checkTruncationLength()`, which also doesn't work because we check for additional conditions before truncating.